src/core/mac/data_poll_handler.hpp

```diff
@@ -57,7 +55,11 @@ namespace ot {
  * @{
  */
 
+#if OPENTHREAD_FTD || OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
```

Comment:
For `DataPollHandler`, the only thing we need is its `NeighborInfo` struct definition, which we can include in all builds (no config check). Everything else in `DataPollHandler` is limited to `OPENTHREAD_FTD`. This is better since adding `OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` as a guard check for the "data poll handler" (which has nothing to do with CSL) did not seem good to me.

Regarding `DataPollHandler::NeighborInfo` itself, we need it since it tracks properties for handling retransmission of any indirect message transmission (which would apply to CSL and poll-based indirect tx). We can later move some of these fields into `IndirectSender::NeighborInfo`, and this way, we can fully decouple CSL and the "data poll handler." I may try this later (in a follow-up PR).